content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -867,6 +873,53 @@ is disabled, `.spec.completions` is immutable.
 Use cases for elastic Indexed Jobs include batch workloads which require 
 scaling an indexed Job, such as MPI, Horovord, Ray, and PyTorch training jobs.
 
+### Delayed creation of replacement pods
+
+{{< feature-state for_k8s_version="v1.28" state="alpha" >}}
+
+{{< note >}}
+You can only set `podReplacementPolicy` on Jobs if you enable the `JobPodReplacementPolicy` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
+{{< /note >}}
+
+By default, the Job controller recreates Pods as soon they either fail or are terminating (have a deletion timestamp).
+This means that, at a given time, when some of the Pods are terminating, the number of running Pods for the Jobs can be greater than `parallelism` or greater than one Pod per index (if using Indexed Jobs).
+
+You may choose to create replacement Pods only when the terminating Pod is fully terminal (has `status.phase: Failed`). To do this, set the `.spec.podReplacementPolicy: Failed`.
```

Comment:
```suggestion
You may choose to create replacement Pods only when the terminating Pod is fully terminal (`status.phase` has a value` Failed` or `Succeeded`). To do this, set the `.spec.podReplacementPolicy: Failed`.
```

Reply:
I don't think this is correct.  I don't know if you can actually have a terminating pod that is also succeeded.  I think the recreation would only happen if a pod is terminal with Failed.

Reply:
For Pods that succeed, `.spec.podReplacementPolicy` makes no difference. We could fix this by adding a glossary entry for “Pod termination” and referring to that. Then, we can omit the text in parens here.

Reply:
> I don't think this is correct. I don't know if you can actually have a terminating pod that is also succeeded. I think the recreation would only happen if a pod is terminal with Failed.

Correct, when `podReplacementPolicy: Failed` we only recreate pods when the previous is terminal AND failed. When `podReplacementPolicy: TerminatingOrFailed` (default) we also recreate pods when they are terminating, and they still may end up in `succeeded` state. This is confusing, but I think a little off here.

> For Pods that succeed, .spec.podReplacementPolicy makes no difference. 

Correct. 

> We could fix this by adding a glossary entry for “Pod termination” and referring to that. Then, we can avoid adding the text in parens here.

For now I would suggest to just rephrase it as "Pod is fully terminal in the `Failed` phase".

Reply:
I had `completions > 1` in mind, which creates a new pod on success. I suppose it depends if we want to describe more generic behaviour of the controller, or if we want to make it specific to the feature.

These sentences evoke the generic behaviour:
- `By default, the Job controller recreates Pods as soon they either fail or are terminating (have a deletion timestamp).`
- `create replacement Pods only when`

> I don't think this is correct. I don't know if you can actually have a terminating pod that is also succeeded. I think the recreation would only happen if a pod is terminal with Failed.

yeah, I guess my suggestion makes it confusing, if by terminating pods we mean pods that they have received a termination signal.

>  When podReplacementPolicy: TerminatingOrFailed (default) we also recreate pods when they are terminating, and they still may end up in succeeded state. This is confusing, but I think a little off here.

we circumvent the problem here by defining it via deletionTimestamp

> For now I would suggest to just rephrase it as "Pod is fully terminal in the Failed phase".

+1, as it doesn't try to define terminal

Reply:
okay